vmc/resource_vmc_cluster.go

```diff
@@ -0,0 +1,233 @@
+
+	connector := m.(*ConnectorWrapper)
+	orgID := m.(*ConnectorWrapper).OrgID
+	clusterClient := sddcs.NewDefaultClustersClient(connector)
+
+	clusterConfig := &model.ClusterConfig{
+		NumHosts:          numHosts,
+		HostCpuCoresCount: &hostCPUCoresCount,
+		// To be added : support for other host instance types
+		//HostInstanceType:  &hostInstanceType,
+		//StorageCapacity:   &storageCapacityConverted,
+	}
+
+	task, err := clusterClient.Create(orgID, sddcID, *clusterConfig)
+	if err != nil {
+		return fmt.Errorf("error while creating cluster for SDDC with id %s: %v", sddcID, err)
+	}
+
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		tasksClient := orgs.NewDefaultTasksClient(connector)
+		task, err := tasksClient.Get(orgID, task.Id)
+		if err != nil {
+			if err.Error() == (errors.Unauthenticated{}.Error()) {
+				log.Print("Auth error", err.Error(), errors.Unauthenticated{}.Error())
+				err = connector.authenticate()
+				if err != nil {
+					return resource.NonRetryableError(fmt.Errorf("authentication error from Cloud Service Provider : %s", err))
+				}
+				return resource.RetryableError(fmt.Errorf("instance creation still in progress"))
+			}
+			return resource.NonRetryableError(fmt.Errorf("error describing instance: %s", err))
+
+		}
+		if task.Params.HasField(ClusterIdFieldName) {
+			clusterID, err := task.Params.String(ClusterIdFieldName)
+			if err != nil {
+				return resource.NonRetryableError(fmt.Errorf("error getting clusterId : %s", err))
+
+			}
+			d.SetId(clusterID)
+		}
+		if *task.Status != "FINISHED" {
+			return resource.RetryableError(fmt.Errorf("expected instance to be created but was in state %s", *task.Status))
+		}
+		return resource.NonRetryableError(resourceClusterRead(d, m))
+	})
+}
+
+func resourceClusterRead(d *schema.ResourceData, m interface{}) error {
+	connector := (m.(*ConnectorWrapper)).Connector
+	clusterID := d.Id()
+	sddcID := d.Get("sddc_id").(string)
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	sddc, err := GetSDDC(connector, orgID, sddcID)
+	log.Printf("SDDC ID : %s", sddcID)
+	if err != nil {
+		if err.Error() == errors.NewNotFound().Error() {
+			log.Printf("SDDC with ID %s not found", sddcID)
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error while getting the SDDC with ID %s,%v", sddcID, err)
+	}
+
+	if *sddc.SddcState == "DELETED" {
+		log.Printf("Unable to retrieve SDDC with ID %s", sddc.Id)
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(clusterID)
+	cluster := map[string]string{}
+	for i := 1; i < len(sddc.ResourceConfig.Clusters); i++ {
+		currentResourceConfig := sddc.ResourceConfig.Clusters[i]
+		if strings.Contains(currentResourceConfig.ClusterId, clusterID) {
+			cluster["cluster_name"] = *currentResourceConfig.ClusterName
+			cluster["cluster_state"] = *currentResourceConfig.ClusterState
+			cluster["host_instance_type"] = *currentResourceConfig.EsxHostInfo.InstanceType
+			cluster["cluster_id"] = currentResourceConfig.ClusterId
```

Comment:
is it required to set cluster_id in cluster info since id is already set in line 145 above